libgadget/blackhole.c

```diff
@@ -592,10 +581,14 @@ blackhole_feedback_ngbiter(TreeWalkQueryBHFeedback * I,
     if(winds_is_particle_decoupled(other))
         return;
 
-    if(P[other].Swallowed && P[other].Type == 5)	/* we have a black hole merger */
+     /* we have a black hole merger! */
+    if(P[other].Type == 5 && BHP(other).SwallowID != (MyIDType) -1)
     {
         if(BHP(other).SwallowID != I->ID) return;
 
+        /* Swallow the particle*/
+        BHP(other).SwallowTime = All.Time;
+        P[other].Swallowed = 1;
```

Comment:
The distinguish between SwallowID != -1 and Swallowed is not clear (even before the PR).

Reply:
I will try to improve this, at least with a comment

Reply:
Swallowid is set when swallowing is in progress, swallowed is set when swallowing completes. Things gravitate up to swallowing being completed.

Reply:
Thanks. This makes sense -- 'Swallowed' is after it is done. Perhaps SwallowingID? ;)

Reply:
hehe :-)